### DISCUSSION TITLE
Setting speed to zero after final waypoint is reached

### DISCUSSION THREAD

**Discussion by Dr-Goodbeard** [TUDelft-CNS-ATM/bluesky#464] [Q&A] [answered]
Hi, I'm new to using bluesky. I'm wanting to run a scenario where some aircraft are generated and go to their waypoints. That much I have. But then after they reach the last waypoint I want their speed to be set to zero and I want them to stay at that position.

I have something like this at the moment:

00:00:00.00> trail on
00:00:00.00> cre k1 b737 53 -30 0 3000 245
00:00:00.00> orig k1 53 -30
00:00:00.00> dest k1 54 -29
00:00:00.00> pan k1
00:00:01.00> addwpt k1 53 -32

Additionally, how do I get the purple waypoint lines to show without clicking the screen.
Any help is much appreciated

**Comment by ProfHoekstra** [comment, marked as answer]
Indeed, that’s how I would do it as well. There are more AT commands like ATDIST that can be used as well, even with waypoints/positions which are not in the route.

Best regards,
    Jacco Hoekstra


On 7 Jun 2023, at 04:06, nl22-nmsu ***@***.***> wrote:

﻿

Figure this out, but I'll leave posted for the other new comers like me.

00:00:00.00> cre k1 b737 53 -30 0 3000 245
00:00:00.00> pan k1
00:00:01.00> addwpt k1 53 -35
00:00:02.00> k1 at k1001 do k1 spd 0

**Reply by Dr-Goodbeard** [reply]
I had initially thought I figured it out but later deleted that comment because it didn't work for multiple waypoints. 

When I add multiple waypoints and tell it to stop at the last one it either stops at the first waypoints instead or it keeps flying past the last waypoint saying the last waypoint wasn't recognized in the output window. 

I had something like:

00:00:00.00> cre k1 b737 53 -30 0 3000 245 

00:00:00.00> pan k1 

00:00:01.00> addwpt k1 53 -35 

00:00:01.00> addwpt k1 50 -35

00:00:02.00> k1 at k1002 do k1 spd 0 


The output was something like "k1002 is not a waypoint"

**Comment by jooste** [comment]
Hi @nl22-nmsu, are you sure you didn't make a typo? When I enter the same commands I get no error.

**Reply by Dr-Goodbeard** [reply]
Weird, last night that stream of commands wasn't working but joe it does. Thank you for you time though!

And I'm not sure if I should open a new discussion for this, but is there a command to turn on the purple lines that show the waypoints.
